### ISSUE TITLE
Support for Concepts TS (not merely documented concepts)

### ISSUE THREAD

**Issue by h-2** [foonathan/standardese#58]
[I didn't want to piggyback on the other issue because this is about actual concepts, not just doc]

We are currently writing a library that uses actual concepts (as in the TS) and I was wondering if standardese handles them any better than regular doxygen?
Doxygen makes regular templatized variables of the concept definitions (which is not nice, but ok) and it puts **all** of the requires clause into the return type of a constrained functions (which makes the documentation horrible to read).
Does standardese work any better or is it limited by libclang's inability to parse Concepts TS right now?

Thanks!

**Comment by foonathan** [comment]
Yes, it is also limited by libclang. I might be able to workaround that somehow; I'm currently extracting the parsing code in a separate library, will be released this week (hopefully).

**Comment by h-2** [comment]
Thanks for the quick reply!

> I might be able to workaround that somehow; I'm currently extracting the parsing code in a separate library,

Cool, if there is anything testable in regards to concepts and you need feedback, don't hesitate to let me know!

**Comment by foonathan** [comment]
You can check out https://github.com/foonathan/cppast and see how it handles concepts (it has a simple tool that dumps the AST).

**Comment by wovo** [comment]
What is the status of concepts? Not supported because clang doesn't support concepts?

**Comment by wovo** [comment]
I dug up somewhere that the under-development clang support for concepts is activated by -fconcepts-ts, but the LLVM/clang I downloaded doesn't accept that option. A pity the code I work on relies heavily on concepts, and it makes perferct sense to put the documentation at the concepts.

**Comment by foonathan** [comment]
Yes, I can't support them because libclang doesn't support them.
